test/e2e/upgrade_clusterctl_test.go

```diff
@@ -377,6 +385,20 @@ func preUpgrade(clusterProxy framework.ClusterProxy, bmoUpgradeToRelease string,
 	Expect(err).NotTo(HaveOccurred())
 }
 
+// postUpgrade hook is for installing the new Metal3 IPAM provider
+// when upgrading from CAPM3 bundled IPAM.
+func postUpgrade(ctx context.Context, clusterProxy framework.ClusterProxy) {
+	By("Installing Metal3 IPAM provider")
+	ipamDeployLogFolder := filepath.Join(os.TempDir(), "target_cluster_logs", "ipam-deploy-logs", clusterProxy.GetName())
+	input := clusterctl.InitInput{
+		ClusterctlConfigPath: clusterctlConfigPath,
+		KubeconfigPath:       clusterProxy.GetKubeconfigPath(),
+		LogFolder:            ipamDeployLogFolder,
+		IPAMProviders:        []string{Metal3ipamProviderName},
```

Comment:
```suggestion
		IPAMProviders:        []string{fmt.Sprintf(Metal3ipamProviderName, ipamRelease)},
```
We need to give the version here.